sklearn/cross_decomposition/_pls.py

```diff
@@ -549,6 +603,13 @@ class PLSRegression(_PLS):
         and potentially scaling. If `False`, these operations will be done
         inplace, modifying both arrays.
 
+    algorithm : {'nipals', 'svd', 'dayalmacgregor', 'kernel'}, default='nipals'
+        The algorithm used to estimate the first singular vectors of the
+        cross-covariance matrix. 'nipals' uses the power method while 'svd'
+        will compute the whole SVD. 'dayalmacgregor' and 'kernel' are
+        equivalent, and calculate the `X` score, weight and rotation
+        matrices without deflating `Y`.
```

Comment:
I would rather have a single option named `algorithm="dayalmacgregor"` and just mention in the docstring that this is also sometimes referred to as the "kernel" algorithm.

```suggestion
    algorithm : {'nipals', 'svd', 'dayalmacgregor'}, default='nipals'
        The algorithm used to estimate the first singular vectors of the
        cross-covariance matrix. 'nipals' uses the power method while 'svd'
        will compute the whole SVD. 'dayalmacgregor' (sometimes also known
        as 'kernel' is generally faster than NIPALS and calculates the
        `X` score, weight and rotation matrices without deflating `Y`.
```

Reply:
Could you also please add a "References" section near the bottom of this docstring with links to the reference papers (both for NIPALS and Dayal-Mac Gregor)?

Please have a look a the docstrings of other scikit-learn estimators to learn how we typically order sections in a docstring and how we typically format bibliographical references.

You can use:

```
git grep -C 5 References sklearn/
```

to quickly find examples.

Reply:
makes sense to me. I removed all instances of "kernel" and now it's just "dayalmacgregor".

will add the Dayal-MacGregor paper reference with a `:doi:` sphinx tag as well in a follow up commit

Reply:
done, references added for NIPALS and DayalMacGregor